src/instructlab/model/upload.py

```diff
@@ -158,3 +198,80 @@ def upload_safetensors(self) -> None:
         logger.info(
             f"\nUploading safetensors model at {self.local_model_path} succeeded!"
         )
+
+
+class OCIModelUploader(ModelUploader):
+    """
+    Class to handle uploading OCI-formatted models to an OCI registry endpoint
+    We are leveraging OCI v1.1 for this functionality
+    """
+
+    def __init__(
+        self,
+        model: str,
+        destination: str,
+        release: str,
+    ) -> None:
+        super().__init__(
+            model=model,
+            destination=destination,
+            release=release,
+        )
+
+    def upload(self):
+        """
+        Upload specified model to the specified OCI registry
+        """
+
+        logger.info(
+            f"Uploading model to OCI registry:\n{DEFAULT_INDENT}Model: {self.model}\n{DEFAULT_INDENT}Destination: {self.destination}@{self.release}\n"
+        )
+
+        # determine if model is path or name - look in checkpoints dir if the latter
+        if os.path.exists(self.model):
+            self.local_model_path = Path(self.model)
+            self.model = self.model.split("/")[-1]
+        else:
+            self.local_model_path = os.path.join(DEFAULTS.CHECKPOINTS_DIR, self.model)
+            # throw an error if the built path is invalid
+            if not os.path.exists(self.local_model_path):
+                raise FileNotFoundError(
+                    f"Couldn't find model at {self.local_model_path} - are you sure it exists?"
+                )
+
+        # upload OCI model, if valid
+        invalid_model = re.search(r"^(?:[^:]*:){2}(.*)$", str(self.local_model_path))
+        if invalid_model:
+            logger.error(
+                f"\nInvalid model supplied:\n{DEFAULT_INDENT}Please specify tag/version '{invalid_model.group(1)}' via --release",
+            )
+            raise InvalidModelForUpload(self.local_model_path)
+        if not is_oci_repo(self.destination):
+            logger.error(
+                f"\nInvalid destination supplied:\n{DEFAULT_INDENT}Please specify valid OCI repository URL syntax via --destination",
+            )
+            raise InvalidDestinationForUpload(self.destination, self.release)
+        self.upload_oci()
+
+    def upload_oci(self) -> None:
+        # Check if skopeo is installed and the version is at least 1.9
+        check_skopeo_version()
+
+        command = [
+            "skopeo",
+            "copy",
+            f"oci:{self.local_model_path}",
+            f"{self.destination}:{self.release}",
+            "--remove-signatures",
```

Comment:
Why do we remove signatures, out of curiosity?

Reply:
To be honest, I did this because this is what we do for `download` - we could possibly remove this, but maybe we should wait until #821 is complete and we could remove this as part of that work?

Reply:
its a downstream specific reason, skopeo needs this right now to be able to pull from RH registries
so I dont think we can remove this until we move away from using skopeo for uploads and downloads

Reply:
that said it may not be required during upload - i would test without it to be sure

Reply:
Thanks both - I was mostly just curious. My thought was that if a user goes through the trouble to sign their OCI artifact model out of band, we shouldn't strip that away if there's no reason to. But if that breaks downstream, then for sure keep it in.